[Live Range Selection] editing/spelling/spelling-attribute-change.html fails

rniwa opened the pull request:
#### 4ee2b8c5f8a526e95cae00b8074b578b098aef0e
<pre>
[Live Range Selection] editing/spelling/spelling-attribute-change.html fails
<a href="https://bugs.webkit.org/show_bug.cgi?id=248789">https://bugs.webkit.org/show_bug.cgi?id=248789</a>

Reviewed by Wenson Hsieh.

Use a valid offset to set the selection at the end of the target element.

* LayoutTests/editing/spelling/spelling-attribute-change.html:

Canonical link: <a href="https://commits.webkit.org/257391@main">https://commits.webkit.org/257391@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b2304cd2546b176e4d7d97eb216aae1cb0aad32e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98831 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8044 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31979 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108248 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168506 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8594 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85411 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91364 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106224 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104514 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/6511 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90065 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33534 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88333 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21408 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76399 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1956 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22939 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1865 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/45408 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6821 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42396 "Passed tests") | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2570 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3262 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on LayoutTests/editing/spelling/spelling-attribute-change.html:
```diff
@@ -27,7 +27,7 @@
     if (target.tagName == "SPAN") {
         target.appendChild(document.createTextNode("Hello,"));
-        sel.setBaseAndExtent(target, 6, target, 6);
+        sel.setBaseAndExtent(target, target.childNodes.length, target, target.childNodes.length);
```
(Or maybe `setPosition()` would be more concise?)

rniwa committed:
[Live Range Selection] editing/spelling/spelling-attribute-change.html fails
https://bugs.webkit.org/show_bug.cgi?id=248789

Reviewed by Wenson Hsieh.

Use a valid offset to set the selection at the end of the target element.

* LayoutTests/editing/spelling/spelling-attribute-change.html:

Canonical link: https://commits.webkit.org/257391@main
